src/text-input/src/TextInputField.js

```diff
@@ -96,7 +93,7 @@ export default class TextInputField extends PureComponent {
       ...props
     } = this.props
 
-    const { id } = this.state
+    const id = `TextInputField-${this.state.id}`
```

Comment:
This is to avoid collisions with other components using a naive `id` incrementer (cant have the same `id` in a document more than once).